### PR TITLE
Tidyup PaymentMethods (logos) component

### PR DESCRIPTION
**Pull request by lsl**
`web-payment` as an option for PaymentLogo doesn't make sense and it previously added a blank placeholder where a logo would have been shown, this PR removes that option.

Removing `web-payment` as a PaymentLogo option also uncovered a chance to clean the code a little. I hope this is simpler but it at least clarifies/enforces the idea that there is not a 1:1 relationship between PaymentLogo and payment methods.

#### Changes proposed in this Pull Request

* Remove Web Payment as possible logo type
* Alignment fix
* Add proptypes
* noop on empty method set
* No need for class
* Filter out invalid type values
* Rejig PaymentLogo map render
* Clean up brandname alttext for cc's

#### Testing instructions

* Cosmetic, payment logos are shown on the plans page.

#### Before (gap from web-payment)
screenshot_2019-02-26 plans my first site wordpress com
#### After
screenshot_2019-02-26 plans my first site wordpress com 2

#### Before (alignment)
screenshot_2019-02-26 plans my first site wordpress com 3
#### After
screenshot_2019-02-26 plans my first site wordpress com 1

### CODE REVIEW

File: client/components/payment-logo/index.jsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import i18n from 'i18n-calypso';
 
 const ALT_TEXT = {
 	alipay: 'Alipay',
-	amex: 'American Express',
+	amex: 'Amex',
```
Comment (yoavf) [inline client/components/payment-logo/index.jsx]: Why? the logo does say American Express and that's consistent with the branding.

Comment (lsl) [reply client/components/payment-logo/index.jsx]: The `PaymentMethod` was overriding these anyway, figured I'd preserve what was currently being used.

Amex/Visa was a coinflip, discover was a noop, and Mastercard is definitely lower case c now. MasterCard is their old branding.

Comment (yoavf) [reply client/components/payment-logo/index.jsx]: Mastercard/Visa is indeed better, but let's keep American Express for Amex and fix elsewhere if needed.

Comment (lsl) [reply client/components/payment-logo/index.jsx]: (Happy to change)

**Pull request merged**